main/sceneManager.cpp

```diff
@@ -1,26 +1,66 @@
 #include "main/sceneManager.h"
 
-using Tri = objects::TriangleObj;
+SceneManager::SceneManager()
+{
+    createCollectors();
+}
+
+std::vector<objects::EnergyCollector *> SceneManager::getEnergyCollectors()
+{
+    std::vector<objects::EnergyCollector *> temp;
+    temp.reserve(_energyCollectors.size());
+    for (const auto &t : _energyCollectors)
+    {
+        temp.push_back(t.get());
+    }
+    return temp;
+}
+
+std::vector<objects::TriangleObj *> SceneManager::getModelTriangles()
+{
+    std::vector<objects::TriangleObj *> temp;
+    temp.reserve(_modelTriangles.size());
+    for (const auto &t : _modelTriangles)
+    {
+        temp.push_back(t.get());
+    }
+    return temp;
+}
 
-void IamSimulator()
+// PRIVATE METHODS
+
+void SceneManager::createCollectors()
 {
-    SceneManager manager;
-    CHECK(manager.loadModel("123231")) << "Cannot load model";
+    _energyCollectors.reserve(constants::kCollectors);
 
-    std::vector<Tri> triangles = manager.getModelTriangles();
-    generators::PointSource source;
+    const double collectorRadius = static_cast<double>(constants::kSimulationRadius) / 2;
+    double alphaIncrement;
 
-    for (int i = 0; i < 1000000; i++)
+    if (constants::kCollectors % 2 == 0)
+    {
+        alphaIncrement = 2 * constants::kPi * 180 / (static_cast<double>(constants::kCollectors) / 2 * 360);
+    }
+    else
     {
-        auto ray = source.generateRay(1, 2);
+        alphaIncrement = 2 * constants::kPi * 180 / ((static_cast<double>(constants::kCollectors) - 1) / 2 * 360);
+    }
 
-        for (const Tri &triangle : triangles)
+    for (double alpha = 0; alpha <= constants::kPi; alpha += alphaIncrement)
+    {
+        double cur = collectorRadius * std::sin(alpha);
```

Comment:
What does `cur` stand for? Please name it so that there is no confusion.

Reply:
I couldn't come up with the idea for x position and y at the same time... so I used cur from "current" :P do you think that tempNum will be better? ;)

Reply:
Answer yourself. "tempNum" does not say anything except that it is temp. We will know it is a number from the type.

How about: "groundCoordinate", or maybe baseCoordinate? I do not have a good name either, but "cur" is bad.

Reply:
groundCoordinate sounds good 👍